src/pybind/mgr/dashboard/plugins/feature_toggles.py

```diff
@@ -0,0 +1,140 @@
+        RbdMirroringSummary, RbdMirroringPoolMode, RbdMirroringPoolPeer],
+    Features.ISCSI: [Iscsi, IscsiTarget],
+    Features.CEPHFS: [CephFS],
+    Features.RGW: [Rgw, RgwDaemon, RgwBucket, RgwUser],
+}
+
+
+class Actions(Enum):
+    ENABLE = 'enable'
+    DISABLE = 'disable'
+    STATUS = 'status'
+
+
+@PM.add_plugin
+class FeatureToggles(I.CanMgr, I.CanLog, I.Setupable, I.HasOptions,
+                     I.HasCommands, I.FilterRequest.BeforeHandler,
+                     I.HasControllers):
+    OPTION_FMT = 'FEATURE_TOGGLE_{}'
+    CACHE_MAX_SIZE = 128  # Optimum performance with 2^N sizes
+    CACHE_TTL = 10  # seconds
+
+    @PM.add_hook
+    def setup(self):
+        self.Controller2Feature = {
+            controller: feature
+            for feature, controllers in Feature2Controller.items()
+            for controller in controllers}
+
+    @PM.add_hook
+    def get_options(self):
+        return [Option(
+            name=self.OPTION_FMT.format(feature.value),
+            default=(feature not in PREDISABLED_FEATURES),
+            type='bool',) for feature in Features]
+
+    @PM.add_hook
+    def register_commands(self):
+        @CLICommand(
+            "dashboard feature",
+            "name=action,type=CephChoices,strings={} ".format(
+                "|".join(a.value for a in Actions))
+            + "name=features,type=CephChoices,strings={},req=false,n=N".format(
+                "|".join(f.value for f in Features)),
+            "Enable or disable features in Ceph-Mgr Dashboard")
+        def cmd(mgr, action, features=None):
+            ret = 0
+            msg = []
+            if action in [Actions.ENABLE.value, Actions.DISABLE.value]:
+                if features is None:
+                    ret = 1
+                    msg = ["At least one feature must be specified"]
+                else:
+                    for feature in features:
+                        mgr.set_module_option(
+                            self.OPTION_FMT.format(feature),
+                            action == Actions.ENABLE.value)
+                        msg += ["Feature '{}': {}".format(
+                            feature,
+                            'enabled' if action == Actions.ENABLE.value else
+                            'disabled')]
+            else:
+                for feature in features or [f.value for f in Features]:
+                    enabled = mgr.get_module_option(self.OPTION_FMT.format(feature))
+                    msg += ["Feature '{}': '{}'".format(
+                        feature,
+                        'enabled' if enabled else 'disabled')]
+            return ret, '\n'.join(msg), ''
+        return {'handle_command': cmd}
+
+    def _get_feature_from_request(self, request):
+        try:
+            return self.Controller2Feature[
+                request.handler.callable.__self__]
+        except (AttributeError, KeyError):
+            return None
+
+    @ttl_cache(ttl=0, maxsize=CACHE_MAX_SIZE)
```

Comment:
Doesn't this disable the cache?

Reply:
Thanks, great catch! I don't remember why I left that (there's `CACHE_TTL` over there).